hack/build.sh

```diff
@@ -41,55 +41,61 @@ readonly SITE=$PWD/site
   curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/main/docs/serving-api.md -s > "$TEMP/docs-main/docs/serving/reference/serving-api.md"
   curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/main/docs/eventing-api.md -s > "$TEMP/docs-main/docs/eventing/reference/eventing-api.md"
-  pushd "$TEMP/docs-main"; mkdocs build -f mkdocs.yml -d $SITE/development; popd
-
-  # Latest release branch to /docs
-  git clone --depth 1 -b ${DOCS_BRANCHES[0]} https://github.com/${GIT_SLUG} "$TEMP/docs-$latest"
-
-  if [ ${latest#*1.} -gt 6 ]; then
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/${DOCS_BRANCHES[0]}/docs/serving-api.md -s > "$TEMP/docs-$latest/docs/serving/reference/serving-api.md"
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/${DOCS_BRANCHES[0]}/docs/eventing-api.md -s > "$TEMP/docs-$latest/docs/eventing/reference/eventing-api.md"
-  else
-      curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/${DOCS_BRANCHES[0]}/docs/serving-api.md -s > "$TEMP/docs-$latest/docs/reference/api/serving-api.md"
-  curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/${DOCS_BRANCHES[0]}/docs/eventing-api.md -s > "$TEMP/docs-$latest/docs/reference/api/eventing-api.md"
-  fi
-
-  pushd "$TEMP/docs-$latest"; KNATIVE_VERSION="${VERSIONS[0]}.0" SAMPLES_BRANCH="${DOCS_BRANCHES[0]}" mkdocs build -d $SITE/docs; popd
-
-  # Previous release branches release-$version to /v$version-docs
+  
+  # Create docs directory structure
+  mkdir -p "$SITE/docs"
+  
+  # Build latest docs to /docs
+  pushd "$TEMP/docs-main"
+  KNATIVE_VERSION="${VERSIONS[0]}.0" SAMPLES_BRANCH="${DOCS_BRANCHES[0]}" mkdocs build -d "$SITE/docs"
+  popd
+
+  # Build versioned docs to /vX.Y-docs/
+  for i in "${!previous[@]}"; do
+    version="${previous[$i]}"
+    branch="${DOCS_BRANCHES[$((i+1))]}"
+    
+    git clone --depth 1 -b "$branch" "https://github.com/$GIT_SLUG" "$TEMP/docs-$version"
+    
+    # Fetch API reference docs for versioned builds
+    curl -f -L --show-error https://raw.githubusercontent.com/knative/serving/$branch/docs/serving-api.md -s > "$TEMP/docs-$version/docs/serving/reference/serving-api.md"
+    curl -f -L --show-error https://raw.githubusercontent.com/knative/eventing/$branch/docs/eventing-api.md -s > "$TEMP/docs-$version/docs/eventing/reference/eventing-api.md"
+    
+    pushd "$TEMP/docs-$version"
+    KNATIVE_VERSION="$version.0" SAMPLES_BRANCH="$branch" mkdocs build -d "$SITE/v$version-docs"
+    popd
+  done
+  
+  # Build development site
+  pushd "$TEMP/docs-main"
+  KNATIVE_VERSION="${VERSIONS[0]}.0" SAMPLES_BRANCH="${DOCS_BRANCHES[0]}" mkdocs build -f mkdocs.yml -d "$SITE/development"
+  popd
+  
+  # Move non-versioned content to root level
+  mkdir -p "$SITE/about"
+  cp -r "$TEMP/docs-main/docs/about" "$SITE/"
+  mkdir -p "$SITE/community"
+  cp -r "$TEMP/docs-main/docs/community" "$SITE/"
```

Comment:
I think this _may_ break links for images, CSS, and JavaScript for these pages.  I'll attempt another build with your changes this afternoon (I avoided this PR yesterday because I had a few other items and I really got pulled in on Friday).

Reply:
The `index.html` did seem to mostly work with just a copy, mostly because that page is largely hand-crafted HTML.

There are still a few items I'm trying to figure out in the navigation -- like why the "blog", "about", and "community" links don't show up in the navigation tabs, despite being in `nav.html`.